pkg/ext-proc/scheduling/scheduler.go

```diff
@@ -0,0 +1,54 @@
+// Package scheduling implements request scheduling algorithms.
+package scheduling
+
+import (
+	"math/rand"
+
+	klog "k8s.io/klog/v2"
+
+	"ext-proc/backend"
+)
+
+var (
+	defaultFilter = &filter{
```

Comment:
We may want a better way to represent nodes in this state machine. Not a blocker, but worth thinking about in the future since we may have multiple.

Reply:
yeah as we are implementing more advanced algos will see how this will evolve.

Reply:
closing this for now